tests/test_validation.py

```diff
@@ -415,30 +428,136 @@ def test_validation_2bin_2channel_coupledshapefactor():
+                            {
+                                'name': 'coupled_shapefactor',
+                                'type': 'shapefactor',
+                                'data': None
+                            }
                         ]
                     }
                 ]
             }
         ]
     }
-    pdf  = pyhf.hfpdf(spec)
-    data = []
-    for c in pdf.spec['channels']:
-        data += source['channels'][c['name']]['bindata']['data']
-    data = data + pdf.config.auxdata
-
+    return spec
+
+
+@pytest.fixture(scope='module')
+def expected_result_2bin_2channel_coupledshapefactor(mu=1.):
+    if mu == 1:
+        expected_result = {
+            'obs': 0.5421679124909312,
+            'exp': [
+                0.013753299929451691,
+                0.048887400056355966,
+                0.15555296253957684,
+                0.4007561343326305,
+                0.7357169630955912
+            ]
+        }
+    return expected_result
+
+
+def validate_runOnePoint(pdf, data, mu_test, expected_result, tolerance=1e-5):
     init_pars = pdf.config.suggested_init()
     par_bounds = pdf.config.suggested_bounds()
 
-    assert len(pdf.config.auxdata) == 0
-    assert len(init_pars)  == 3 #1 mu 2 shared shapefactors
-    assert len(par_bounds) == 3
-
-    muTest = 1.0
-    clsobs, cls_exp = pyhf.runOnePoint(muTest, data,pdf,init_pars,par_bounds)[-2:]
-    cls_obs = 1./clsobs
-    cls_exp = [1./x for x in cls_exp]
-    assert (cls_obs - expected_result['obs'])/expected_result['obs'] < VALIDATION_TOLERANCE
-    for result,expected_result in zip(cls_exp, expected_result['exp']):
-        assert (result-expected_result)/expected_result < VALIDATION_TOLERANCE
+    CLs_obs, CLs_exp = pyhf.runOnePoint(
+        mu_test, data, pdf, init_pars, par_bounds)[-2:]
+    CLs_obs = 1. / CLs_obs
+    CLs_exp = [1. / x for x in CLs_exp]
+    assert (CLs_obs - expected_result['obs']) / \
+        expected_result['obs'] < tolerance
+    for result, expected_result in zip(CLs_exp, expected_result['exp']):
+        assert (result - expected_result) / \
+            expected_result < tolerance
+
+
+def test_validation_1bin_shapesys(source_1bin_example1):
+    expected_result = {
+        'obs': 0.4541865416107029,
+        'exp': [
+            0.06371799398864626,
+            0.15096503398048894,
+            0.3279606950533305,
+            0.6046087303039118,
+            0.8662627605298466
+        ]
+    }
+
+    source = source_1bin_example1
+    pdf = pyhf.simplemodels.hepdata_like(source['bindata']['sig'],
+                                         source['bindata']['bkg'],
+                                         source['bindata']['bkgerr'])
+
+    data = source['bindata']['data'] + pdf.config.auxdata
+
+    assert len(pdf.config.suggested_init()) == 2
+    assert len(pdf.config.suggested_bounds()) == 2
+    validate_runOnePoint(pdf, data, 1.0, expected_result)
+
+
```

Comment:
@lukasheinrich There is still cleaning and refactoring that needs to be done, but is the below along the lines of what you were thinking? Or by moving everything into one parameterized function did I miss the point?

Reply:
hi @matthewfeickert yes this is exactly what I meant. possibly the fixture can just include both the spec and the expected results as say a dictionary, but no strong opinion on this